network/server_discovery.go

```diff
@@ -196,7 +197,10 @@ func (s *Server) setupDiscovery() error {
 	keyID := kb.ConvertPeerID(s.host.ID())
 
 	routingTable, err := kb.NewRoutingTable(
-		defaultBucketSize,
+		helperCommon.MaxInt(
+			helperCommon.ClampInt64ToInt(s.config.MaxPeers),
```

Comment:
I'm not sure whether bucket size should equal to your max peer length?
Should it be consider as a degree number?

Reply:
`routingTable` storage peer address, I think the number of  `BucketSize` should be matched, I didn't observe this error log after modification